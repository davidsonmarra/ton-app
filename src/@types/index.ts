export type { ProductDTO } from './productDTO';
export type { ProductsStackParamList } from './routes';
