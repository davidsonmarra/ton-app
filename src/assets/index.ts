export { default as T1Img } from './t1.png';
