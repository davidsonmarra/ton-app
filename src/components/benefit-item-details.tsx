import React from 'react';
import styled, { useTheme } from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

interface BenefitItemDetails {
  icon: string;
  description: string;
}

export const BenefitItemDetails = ({ icon, description }: BenefitItemDetails) => {
  const { colors } = useTheme();

  return (
    <StyledContainer>
      <Icon name={icon} size={24} color={colors.primary} />
      <StyledDescription>{description}</StyledDescription>
    </StyledContainer>
  );
};

const StyledContainer = styled.View`
  flex-direction: row;
  align-items: center;
  margin-bottom: 8px;
`;

const StyledDescription = styled.Text`
  flex: 1;
  margin-left: 8px;
  font-family: ${({ theme: { fonts } }) => fonts.primary.regular};
  font-size: ${({ theme: { fontSizes } }) => fontSizes.md};
  color: ${({ theme: { colors } }) => colors.text};
`;
