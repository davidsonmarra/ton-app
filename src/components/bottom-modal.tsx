import React from 'react';
import { ModalProps, View } from 'react-native';
import styled, { useTheme } from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { useSafeAreaInsets } from 'react-native-safe-area-context';

interface BottomModalProps extends ModalProps {
  children: React.ReactNode;
  title: string;
}

export const BottomModal = ({ children, title, ...rest }: BottomModalProps) => {
  const { colors } = useTheme();
  const { bottom } = useSafeAreaInsets();

  return (
    <StyledModal {...rest} animated animationType='fade' testID='bottom-modal'>
      <StyledContainer onPress={rest.onRequestClose}>
        <StyledContent bottomInset={bottom}>
          <StyledTitleContainer>
            <View />
            <StyledTitle>{title}</StyledTitle>
            <StyledIconButton onPress={rest.onRequestClose}>
              <Icon name='close' size={32} color={colors.text} />
            </StyledIconButton>
          </StyledTitleContainer>
          {children}
        </StyledContent>
      </StyledContainer>
    </StyledModal>
  );
};

const StyledModal = styled.Modal.attrs({
  transparent: true,
  animationType: 'fade'
})``;

const StyledContainer = styled.TouchableOpacity.attrs({
  activeOpacity: 1
})`
  flex: 1;
  background-color: #00000055;
`;

const StyledContent = styled.View<{ bottomInset: number }>`
  position: absolute;
  width: 100%;
  padding: 24px 24px ${({ bottomInset }) => bottomInset + 12}px;
  bottom: 0;
  justify-content: center;
  align-items: center;
  background-color: ${({ theme: { colors } }) => colors.background};
`;

const StyledTitleContainer = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 24px;
  width: 100%;
`;

const StyledTitle = styled.Text`
  font-family: ${({ theme: { fonts } }) => fonts.primary.semiBold};
  font-size: ${({ theme: { fontSizes } }) => fontSizes.xl};
  color: ${({ theme: { colors } }) => colors.text};
`;

const StyledIconButton = styled.TouchableOpacity``;
