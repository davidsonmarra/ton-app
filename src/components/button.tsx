import React from 'react';
import { TouchableOpacityProps } from 'react-native/types';
import styled from 'styled-components/native';

interface ButtonProps extends TouchableOpacityProps {
  label: string;
}

export const Button = ({ label, onPress }: ButtonProps) => {
  return (
    <StyledContainer onPress={onPress}>
      <StyledText>{label}</StyledText>
    </StyledContainer>
  );
};

const StyledContainer = styled.TouchableOpacity`
  width: 100%;
  padding: 12px;
  background-color: ${({ theme }) => theme.colors.primary};
  border-radius: 50px;
`;

const StyledText = styled.Text`
  text-align: center;
  font-family: ${({ theme: { fonts } }) => fonts.primary.bold};
  font-size: ${({ theme: { fontSizes } }) => fontSizes.md};
  color: ${({ theme: { colors } }) => colors.secondary};
`;
