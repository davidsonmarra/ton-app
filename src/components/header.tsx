import React from 'react';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import styled from 'styled-components/native';

interface HeaderProps {
  title: string;
}

export const Header = ({ title }: HeaderProps) => {
  const { top } = useSafeAreaInsets();

  return (
    <StyledContainer topInset={top}>
      <StyledTitle>{title}</StyledTitle>
    </StyledContainer>
  );
};

const StyledContainer = styled.View<{ topInset: number }>`
  padding: ${({ topInset }) => topInset + 12}px 12px 12px;
  background-color: ${({ theme: { colors } }) => colors.primary};
`;

const StyledTitle = styled.Text`
  text-align: center;
  font-family: ${({ theme: { fonts } }) => fonts.primary.bold};
  font-size: ${({ theme: { fontSizes } }) => fontSizes.lg};
  color: ${({ theme: { colors } }) => colors.secondary};
`;
