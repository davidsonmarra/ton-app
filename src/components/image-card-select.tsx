import React from 'react';
import styled from 'styled-components/native';

interface ImageCardSelectProps {
  img: string;
  isSelected: boolean;
  handlePressCard: (img: string) => void;
}

export const ImageCardSelect = ({ img, isSelected, handlePressCard }: ImageCardSelectProps) => {
  return (
    <StyledContainer
      style={{
        shadowColor: '#000',
        shadowOffset: {
          width: 0,
          height: 4
        },
        shadowOpacity: 0.32,
        shadowRadius: 5.46,

        elevation: 9
      }}
      isSelected={isSelected}
      onPress={() => handlePressCard(img)}
    >
      <StyledImage source={{ uri: img }} resizeMode='contain' />
    </StyledContainer>
  );
};

const StyledContainer = styled.TouchableOpacity<{ isSelected: boolean }>`
  padding: 12px;
  width: ${250 / 3.5}px;
  height: ${350 / 3.5}px;
  background-color: ${({ theme: { colors } }) => colors.secondary};

  border-radius: 8px;
  border: 2px solid
    ${({ theme: { colors }, isSelected }) => (isSelected ? colors.primary : 'transparent')};
`;

const StyledImage = styled.Image`
  flex: 1;
`;
