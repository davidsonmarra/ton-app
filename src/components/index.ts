export { Button } from './button';
export { BottomModal } from './bottom-modal';
export { BenefitItemDetails } from './benefit-item-details';
export { Header } from './header';
export { ImageCardSelect } from './image-card-select';
export { ImageDetails } from './image-details';
export { PriceDetails } from './price-details';
export { ProductCard } from './product-card';
export { TitleProductDetails } from './title-product-details';
