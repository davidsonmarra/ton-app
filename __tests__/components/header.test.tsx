import { Header } from '../../src/components';
import { matchSnapshotWithProviders, render, fireEvent } from '../test-utils';

const mockHandlePressLeftIcon = jest.fn();
const mockHandlePressRightIcon = jest.fn();

const props = (
  title?: string,
  leftIcon = '',
  rightIcon = '',
  handlePressLeftIcon = mockHandlePressLeftIcon,
  handlePressRightIcon = mockHandlePressRightIcon
) => ({
  title,
  leftIcon,
  rightIcon,
  handlePressLeftIcon,
  handlePressRightIcon
});

describe('Header', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match a snapshot', () => {
    matchSnapshotWithProviders(<Header {...props()} />);
  });

  it('should render correctly with no icons', () => {
    const { queryByTestId } = render(<Header {...props()} />);
    expect(queryByTestId('header-left-icon')).toBeNull();
    expect(queryByTestId('header-right-icon')).toBeNull();
  });

  it('should render correctly with left icon and call function on press', () => {
    const { getByTestId } = render(<Header {...props('title', 'arrow-left')} />);
    expect(getByTestId('header-left-icon')).toBeTruthy();
    fireEvent.press(getByTestId('header-left-button'));
    expect(mockHandlePressLeftIcon).toHaveBeenCalledTimes(1);
  });

  it('should render correctly with right icon and call function on press', () => {
    const { getByTestId } = render(<Header {...props('title', '', 'arrow-right')} />);
    expect(getByTestId('header-right-icon')).toBeTruthy();
    fireEvent.press(getByTestId('header-button-right'));
    expect(mockHandlePressRightIcon).toHaveBeenCalledTimes(1);
  });

  it('should render correctly with both icons and call functions on press', () => {
    const { getByTestId } = render(<Header {...props('title', 'arrow-left', 'arrow-right')} />);
    expect(getByTestId('header-left-icon')).toBeTruthy();
    expect(getByTestId('header-right-icon')).toBeTruthy();
    fireEvent.press(getByTestId('header-left-button'));
    fireEvent.press(getByTestId('header-button-right'));
    expect(mockHandlePressLeftIcon).toHaveBeenCalledTimes(1);
    expect(mockHandlePressRightIcon).toHaveBeenCalledTimes(1);
  });
});
