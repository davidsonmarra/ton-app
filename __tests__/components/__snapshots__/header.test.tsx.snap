// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1Ec677",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
  topInset={0}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": 32,
      }
    }
    testID="header-left-button"
  />
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "Poppins-Bold",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": 32,
      }
    }
    testID="header-button-right"
  />
</View>
`;
